File: ABC298/c.cpp
#include<iostream>
#include<vector>
#include<set>
using namespace std;
int main(){
    int n, Q;
    cin >> n >> Q;

    vector<set<int>> cards(200001);
    vector<multiset<int>> boxes(n+1);
    
    for(int k=0; k<Q; k++){
        int q, i, j;
        cin >> q;
        if(q==1){
            cin >> i >> j;
            cards[i].insert(j);
            boxes[j].insert(i);
        }else if(q==2){
            cin >> i;
            for(auto c:boxes[i]) cout << c << " ";
            cout << endl;
        }else{
            cin >> i;
            for(auto b:cards[i]) cout << b << " ";
            cout << endl;
        }
    }

    return 0;
}

File: ABC298/c.py
n = int(input())
Q = int(input())
cards = [[] for _ in range(200001)]
boxes = [[] for _ in range(n+1)]

for _ in range(Q):
    query = input()
    if(query[0] == "1"):
        q, i, j = map(int, query.split())
        cards[i].append(j)
        boxes[j].append(i)
    if(query[0] == "2"):
        q, i = map(int, query.split())
        boxes[i].sort()
        print(*boxes[i])
    if(query[0] == "3"):
        q, i = map(int, query.split())
        set_cards = set(cards[i])
        cards[i] = list(set_cards)
        cards[i].sort()
        print(*cards[i])